solutions/usaco-944.mdx

```diff
@@ -98,5 +98,88 @@ public class FencePlan {
 ```
   
 </JavaSection>
+
+<CPPSection>
+
+We can take a DFS (depth-first search) approach to this problem. Each moo network 
+is equivalent to a connected component within the graph of the cows. We can 
+create a list that holds all connected components within the graph, and the cows 
+in each one. Then, to find our answer, we can iterate through each of our connected 
+components and calculate the minimum perimeter of a fence needed to encapsulate 
+all of the cows within that connected component. 
+
+```cpp
+
+#include <fstream>
+#include <algorithm>
+#include <vector>
+#define f first
+#define s second
+using namespace std;
+
+ifstream fin("fenceplan.in");
+ofstream fout("fenceplan.out");
+
+int n, m;
+
+vector<int> adj[100001];
```

Comment:
put 1e5 in a constant called `MAX_N`
and then when declaring this array use `MAX_N + 1`
or you could be a normal person and use nested vectors